doc/user/ospfd.rst

```diff
@@ -806,25 +806,19 @@ Showing Information
    Json o/p of this command covers base route information
    i.e all LSAs except opaque lsa info.
 
-.. clicmd:: show ip ospf [vrf <NAME|all>] database [json]
+.. clicmd:: show ip ospf [vrf <NAME|all>] database [self-originate] [json]
 
-.. clicmd:: show ip ospf [vrf <NAME|all>] database (asbr-summary|external|network|router|summary) [json]
-
-.. clicmd:: show ip ospf [vrf <NAME|all>] database (asbr-summary|external|network|router|summary) LINK-STATE-ID [json]
-
-.. clicmd:: show ip ospf [vrf <NAME|all>] database (asbr-summary|external|network|router|summary) LINK-STATE-ID adv-router ADV-ROUTER [json]
-
-.. clicmd:: show ip ospf [vrf <NAME|all>] database (asbr-summary|external|network|router|summary) adv-router ADV-ROUTER [json]
+   Show the OSPF database summary.
 
-.. clicmd:: show ip ospf [vrf <NAME|all>] database (asbr-summary|external|network|router|summary) LINK-STATE-ID self-originate [json]
+.. clicmd:: show ip ospf [vrf <NAME|all>] database max-age [json]
 
-.. clicmd:: show ip ospf [vrf <NAME|all>] database (asbr-summary|external|network|router|summary) self-originate [json]
+   Show all MaxAge LSAs present in the OSPF link-state database.
 
-.. clicmd:: show ip ospf [vrf <NAME|all>] database max-age [json]
+.. clicmd:: show ip ospf [vrf <NAME|all>] database (asbr-summary|external|network|router|summary|nssa-external|opaque-link|opaque-area|opaque-as) [LINK-STATE-ID] [adv-router A.B.C.D] [json]
 
-.. clicmd:: show ip ospf [vrf <NAME|all>] database self-originate [json]
+.. clicmd:: show ip ospf [vrf <NAME|all>] database (asbr-summary|external|network|router|summary|nssa-external|opaque-link|opaque-area|opaque-as) [LINK-STATE-ID] [self-originate] [json]
 
-   Show the OSPF database summary.
+   Show detailed information about the OSPF link-state database.
```

Comment:
It looks , all these commands re-designed as single command right ? then why do we add as multiple commands in documentation ? any specific reason .

Reply:
@rgirada because the command definition is too complex to be presented in documentation context:
```
       show ip ospf [(1-65535)] [vrf <NAME|all>] database\
         [<\
           max-age\
           |self-originate\
           |<\
             detail\
             |<asbr-summary|external|network|router|summary|nssa-external|opaque-link|opaque-area|opaque-as>\
            >\
            [{\
              A.B.C.D\
              |<adv-router A.B.C.D|self-originate>\
            }]\
         >]\
         [json]
```
By breaking down the command into smaller ones, the documentation becomes more accessible I think. I mostly preserved the existing structure and added options and combinations that were missing.